### PR TITLE
Feature/rb 59 parallax system

### PR DESCRIPTION
**Pull request by KitetsuK**
PATCH

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [ ] Ready to merge / Waiting a reviwer to see my work.
- [x] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: assets/Json/parallaxData.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            "spritePath": "assets/Map/preview.png",
+            "width": 100,
+            "height": 100,
+            "position": {
+                "x": -100,
+                "y": 0
+            }
+        }
+    ]
+}
```
Comment (guillaumeAbel) [inline assets/Json/parallaxData.json:22]: EOF

Comment (guillaumeAbel) [inline assets/Json/parallaxData.json]: EOF

File: assets/Json/playerData.json [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                {
+                    "x": 250.0,
+                    "y": 140.0,
+                    "width": 18.0,
+                    "height": 12.0
+                }
+            ]
+        }
+    }
+}
```
Comment (guillaumeAbel) [inline assets/Json/playerData.json]: EOF

File: deps/json/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.15)
+
+include(FetchContent)
+
+FetchContent_Declare(json URL https://github.com/nlohmann/json/releases/download/v3.11.2/json.tar.xz)
+FetchContent_MakeAvailable(json)
+
+target_link_libraries(${PROJECT_NAME_CLIENT} PRIVATE nlohmann_json::nlohmann_json)
+target_link_libraries(${PROJECT_NAME_SERVER} PRIVATE nlohmann_json::nlohmann_json)
```
Comment (guillaumeAbel) [inline deps/json/CMakeLists.txt]: EOF

**Review by guillaumeAbel — CHANGES_REQUESTED** [review]

File: src/ECS/Systems/Managers/SystemManager.cpp [hunk 4]
```diff
@@ -7,6 +7,8 @@
 
 #include "SystemManager.hpp"
 
+#include <iostream>
```
Comment (Saverio976) [inline src/ECS/Systems/Managers/SystemManager.cpp]: iostream ? Logger existe

File: src/ECS/Systems/Managers/SystemManager.hpp [hunk 5]
```diff
@@ -26,13 +26,14 @@ namespace Systems {
         private:
             std::vector<std::function<void(std::size_t, std::size_t)>>&
             getSystems();
-            void setModifiedSystems();
 
             // NOLINTBEGIN(cppcoreguidelines-avoid-non-const-global-variables)
             static std::size_t _managerNb;
+
```
Comment (Saverio976) [inline src/ECS/Systems/Managers/SystemManager.hpp]: pourquoi un espace ?

File: src/ECS/Systems/Systems.cpp [hunk 6]
```diff
@@ -84,6 +88,55 @@ namespace Systems {
...
+        nlohmann::json jsonData = {};
+
+        if (fileData.is_open()) {
+            input << fileData.rdbuf();
+            if (nlohmann::json::accept(input.str())) {
+                jsonData = nlohmann::json::parse(input.str());
+                return jsonData;
+            }
+        }
+        std::cout << "Could not load the json data : An error occured." << std::endl;
```
Comment (Saverio976) [inline src/ECS/Systems/Systems.cpp]: ???? Logger

File: src/ECS/Systems/Systems.cpp [hunk 7]
```diff
@@ -84,6 +88,55 @@ namespace Systems {
...
+        );
+        Registry::getInstance().getComponents<Types::Parallax>().insertBack(
+            {}
+        );
+        Registry::getInstance().setToBackLayers(id);
+    }
+
+    void initParalax(std::size_t managerId, std::size_t systemId)
+    {
+        nlohmann::json jsonData = openJsonData("assets/Json/parallaxData.json");
```
Comment (Saverio976) [inline src/ECS/Systems/Systems.cpp]: const string

**Review by Saverio976 — CHANGES_REQUESTED** [review]

File: src/ECS/Systems/Systems.cpp [hunk 8]
```diff
@@ -7,6 +7,10 @@
 
 #include "Systems.hpp"
 #include <cstddef>
+#include <fstream>
+#include <iostream>
```
Comment (guillaumeAbel) [inline src/ECS/Systems/Systems.cpp]: iostream

**Review by guillaumeAbel — CHANGES_REQUESTED** [review]

**Review by guillaumeAbel — APPROVED** [review]

**Pull request merged**